pkg/products/marin3r/reconciler_test.go

```diff
@@ -233,3 +237,220 @@ func strPtr(str string) *string {
+			serverClient: func() k8sclient.Client {
+				return fakeclient.NewFakeClientWithScheme(scheme, &v1alpha1.EnvoyConfigRevisionList{
+					Items: []v1alpha1.EnvoyConfigRevision{
+						{
+							ObjectMeta: v1.ObjectMeta{
+								Finalizers: []string{
+									"envoyconfigrevisions.marin3r.3scale.net",
+								},
+								Labels: map[string]string{
+									"marin3r.3scale.net/envoy-api": "v2",
+								},
+								Namespace:         "redhat-rhoam-3scale",
+								Name:              "apicast-ratelimit",
+								DeletionTimestamp: &v1.Time{time.Now()},
+							},
+							TypeMeta: v1.TypeMeta{
+								APIVersion: "V2",
+							},
+							Spec:   v1alpha1.EnvoyConfigRevisionSpec{},
+							Status: v1alpha1.EnvoyConfigRevisionStatus{},
+						},
+					},
+				})
+			},
+			installation: &integreatlyv1alpha1.RHMI{
+				Spec: integreatlyv1alpha1.RHMISpec{
+					NamespacePrefix: "redhat-rhoam-",
+				},
+			},
+			want:         integreatlyv1alpha1.PhaseCompleted,
+			testFunction: confirmThatConfigRevisionsAreCleared,
+		},
+		{
+			name: "Confirm that V2 has been deleted while V3 remains",
+			serverClient: func() k8sclient.Client {
+				return fakeclient.NewFakeClientWithScheme(scheme, &v1alpha1.EnvoyConfigRevisionList{
+					Items: []v1alpha1.EnvoyConfigRevision{
+						{
+							ObjectMeta: v1.ObjectMeta{
+								Finalizers: []string{
+									"envoyconfigrevisions.marin3r.3scale.net",
+								},
+								Labels: map[string]string{
+									"marin3r.3scale.net/envoy-api": "v2",
+								},
+								Namespace: "redhat-rhoam-3scale",
+								Name:      "apicast-ratelimit",
+							},
+							TypeMeta: v1.TypeMeta{
+								APIVersion: "V2",
+							},
+							Spec:   v1alpha1.EnvoyConfigRevisionSpec{},
+							Status: v1alpha1.EnvoyConfigRevisionStatus{},
+						},
+						{
+							ObjectMeta: v1.ObjectMeta{
+								Finalizers: []string{
+									"envoyconfigrevisions.marin3r.3scale.net",
+								},
+								Labels: map[string]string{
+									"marin3r.3scale.net/envoy-api": "v2",
+								},
+								Namespace: "redhat-rhoam-3scale",
+								Name:      "apicast-ratelimit2",
+							},
+							TypeMeta: v1.TypeMeta{
+								APIVersion: "V2",
+							},
+							Spec:   v1alpha1.EnvoyConfigRevisionSpec{},
+							Status: v1alpha1.EnvoyConfigRevisionStatus{},
+						},
+						{
+							ObjectMeta: v1.ObjectMeta{
+								Finalizers: []string{
+									"envoyconfigrevisions.marin3r.3scale.net",
+								},
+								Labels: map[string]string{
+									"marin3r.3scale.net/envoy-api": "v3",
+								},
+								Namespace: "redhat-rhoam-3scale",
+								Name:      "apicast-ratelimitv3",
+							},
+							TypeMeta: v1.TypeMeta{
+								APIVersion: "V3",
+							},
+							Spec:   v1alpha1.EnvoyConfigRevisionSpec{},
+							Status: v1alpha1.EnvoyConfigRevisionStatus{},
+						},
+					},
+				})
+			},
+			installation: &integreatlyv1alpha1.RHMI{
+				Spec: integreatlyv1alpha1.RHMISpec{
+					NamespacePrefix: "redhat-rhoam-",
```

Comment:
What do you think about adding a case for other namespace prefixes that we anticipate used with the operator to protect from a change on that line in the future?

Reply:
Done, for `rhoami` ns.